src/sys/user.rs

```diff
@@ -0,0 +1,177 @@
+
+
+#[cfg(any(target_arch = "x86_64"))]
+use libc::{c_ushort, c_uint, c_char, c_int, c_longlong, c_ulonglong};
+#[cfg(any(target_arch = "x86"))]
+use libc::{c_ushort, c_char, c_int, c_long, c_ulong};
+
+use libc::size_t;
+
+pub type Eflag = size_t;
+
+const EFLAG_CARRY: Eflag = 0x1;
+const EFLAG_PARITY: Eflag = 0x1 << 2;
+const EFLAG_AUXILIARY_CARRY: Eflag  =  0x1 << 4;
+const EFLAG_ZERO: Eflag  =  0x1 << 6;
+const EFLAG_SIGN: Eflag  =  0x1 << 7;
+const EFLAG_TRAP: Eflag  =  0x1 << 8;
+const EFLAG_INTERRUPT_ENABLE: Eflag  =  0x1 << 9;
+const EFLAG_DIRECTION: Eflag  =  0x1 << 10;
+const EFLAG_OVERFLOW: Eflag  =  0x1 << 11;
+const EFLAG_IOPRIVILEGE_1: Eflag  =  0x1 << 12;
+const EFLAG_IOPRIVILEGE_2: Eflag  =  0x1 << 13;
+const EFLAG_NESTEDTASK: Eflag  =  0x1 << 14;
+const EFLAG_RESUME: Eflag  =  0x1 << 16;
+const EFLAG_VIRTUAL_80086_MODE: Eflag  =  0x1 << 17;
+const EFLAG_ALIGNMENT_CHECK: Eflag  =  0x1 << 18;
+const EFLAG_VIRTUAL_INTERRUPT: Eflag  =  0x1 << 19;
+const EFLAG_VIRTUAL_INTERRUPT_PENDING: Eflag  =  0x1 << 20;
+const EFLAG_CPUID: Eflag  =  0x1 << 21;
+
+
+#[repr(C)]
+#[cfg(any(target_arch = "x86_64"))]
```

Comment:
This should just be `#[cfg(target_arch = "x86_64")]`. You should make the same change for the rest of these in this file.